runtime/lua/vim/fs.lua

```diff
@@ -334,30 +334,143 @@ function M.find(names, opts)
   return matches
 end
 
+--- Split a Windows path into a prefix and a body, such that the body can be processed like a POSIX
+--- path.
+---
+--- Does not check if the path is a valid Windows path. Invalid paths will give invalid results.
+---
+--- Examples:
+--- - `\\.\C:\foo\bar` -> `\\.\C:`, `\foo\bar`
```

Comment:
Why does it not immediately convert `\` to `/` (or require a "/" path as input)? That would simplify parts of the logic.

Reply:
Ideally, we want this to be used outside of `normalize` as well (for example in `abspath`), and those functions would not normalize or change the slashes, so I explicitly chose against correcting slashes

Reply:
> Ideally, we want this to be used outside of `normalize` as well (for example it's used in #28187), and those functions would not normalize or change the slashes,

Not worth it. If `abspath()` wants to use `\` slashes for some reason it can always do a `replace('/', '\\')` just before it returns its result. But in fact, users/plugins can do that if they want.

The general policy is: Nvim uses `/` slashes everywhere. That is the policy. We need to stop fiddling around with `\` slashes except for some specific cases for _presentation_.

Reply:
> > Ideally, we want this to be used outside of `normalize` as well (for example it's used in #28187), and those functions would not normalize or change the slashes,
> 
> Not worth it. If `abspath()` wants to use `\` slashes for some reason it can always do a `replace('/', '\\')` just before it returns its result. But in fact, users/plugins can do that if they want.
> 
> The general policy is: Nvim uses `/` slashes everywhere. That is the policy. We need to stop fiddling around with `\` slashes except for some specific cases for _presentation_.

It's not about replacing `/` with `\`. I was told that `abspath` should keep slashes as-is, which means that `/` should remain as `/` and `\` should remain `\` even after being converted to an absolute path. If you think it's okay to convert `\` to `/` in abspath then I don't mind going for it

Reply:
> I was told that `abspath` should keep slashes as-is

That's more of a nice to have and not a requirement. I repeat: Nvim uses "/" slashes. We need to stop fucking around with `\` slashes, it's a waste of time.